src/ipc/handler.c

```diff
@@ -815,6 +817,150 @@ static int ipc_glb_gdb_debug(uint32_t header)
 
 }
 
+#if CONFIG_PROBE
+static inline int ipc_probe_init(uint32_t header)
+{
+	struct sof_ipc_probe_dma_set_params *params = _ipc->comp_data;
+	int dma_set_up = (params->hdr.size - sizeof(*params)) /
+			 sizeof(struct probe_dma);
+
+	trace_ipc("ipc_probe_init()");
+
+	return probe_init(dma_set_up ? params->probe_dma : NULL);
```

Comment:
What is this DMA check for ?

Reply:
cAVS interface allows to initialize probe subsystem with or without extraction enabled. Their interface/implementation distinguishes these by payload provided to IPC, specifically, length of flex array in `struct sof_ipc_probe_dma_set_params`.
Based on IPC payload size reported in header, we calculate whether or not pointer `params->probe_dma` is valid, and if not, we're passing NULL to probe subsystem instead.

Reply:
dma_set_up doesn't being to describe what you explained above.

Maybe dma_info_valid?

Also the test for dma_set_up being zero/non-zero seems flaky, the value could be negative with a bad IPC packet...

Reply:
As for the naming - I've changed to `dma_provided`.
As for bad IPC packets handling, that's what we inherited from cAVS.